Implemented test for capsule `redhat-access-insights-puppet` package availability (BZ1315844)

abalakh opened the pull request:
https://bugzilla.redhat.com/show_bug.cgi?id=1315844
```python
pytest -v tests/foreman/api/test_contentmanagement.py -k test_positive_insights_puppet_package_availability
============================= test session starts ==============================
platform darwin -- Python 2.7.13, pytest-3.2.3, py-1.5.2, pluggy-0.4.0 -- /Users/andrii/workspace/env/bin/python
cachedir: .cache
rootdir: /Users/andrii/workspace/robottelo, inifile:
plugins: xdist-1.20.1, services-1.2.1, mock-1.6.3, forked-0.2, cov-2.5.1
collected 12 items
2017-12-13 05:19:27 - conftest - DEBUG - BZ deselect is disabled in settings


tests/foreman/api/test_contentmanagement.py::CapsuleContentManagementTestCase::test_positive_insights_puppet_package_availability PASSED

============================= 11 tests deselected ==============================
=================== 1 passed, 11 deselected in 6.13 seconds ====================
```

abalakh commented:
We can also check whether we can successfully install that package, i'm just not sure whether it's needed - BZ description states the package wasn't bundled and requested to add it to capsule repo,  so checking whether the package is available for installation should be enough for BZ coverage.

ldjebran commented on tests/foreman/api/test_contentmanagement.py:
```diff
@@ -148,6 +148,32 @@ def update_capsule_download_policy(self, capsule_id, download_policy):
+        :CaseLevel: System
+        """
+        package_name = 'redhat-access-insights-puppet'
+        result = ssh.command(
```
why not:
```python
result = self.capsule_vm.run('yum list available | grep {}'.format(package_name))
```

abalakh replied:
Any way to re-init existing capsule? 🙂 
I don't want to wait for another hour between every single test run, `ssh.command` allows me to comment capsule initialization for debugging purposes.

oshtaier replied:
are you finish with PR?

abalakh replied:
Yep, still, all the tests in this module follow the same style, it'll be a nightmare to debug them otherwise. And even setup class contains instructions on how to reuse existing capsule easily, if we introduce `self.capsule_vm.run()` commands such reuse won't be possible so i'm strongly against using `capsule_vm.run()` here.
Btw i'd rather like to see a refactor of capsule module so we could re-init existing one in the same manner as nailgun entities (smth like `self.capsule = Capsule(id=2, name='example.com')`), that'd unblock us here.

oshtaier replied:
can you briefly explain why it is problem to use `self.capsule_vm.run()`? I can see in setup `cls.capsule_vm.create()`, so it is already created and all time necessary already spent

abalakh replied:
We usually have few hosts with capsule installed dedicated for testing. If some tests are failing on jenkins, or you need to debug something - you can simply pass hostname in setupclass and start executing the tests in a matter of seconds.
If we start using `self.capsule_vm.run` - we'll remove the ability to re-use any existing capsule at all and will be forced to create a new one every single time we want to execute the test.
Test is failing on jenkins and you want to reproduce that failure locally? Cool, wait for 1 hour. Reproduced and now want to use some breakpoints for debugging? Wait another hour for capsule creation. Made a fix but there was a typo and test failed? Another hour. Wanna grab test results for your PR? One more hour.
`run()` command will create us terrible limitations with no benefits at all, and basically `run()` is actually a wrapper for `ssh.command()` which just passes the hostname.

tl;dr - using `vm.run()` will make entire module so enjoyable to work with, as hammer's `test_positive_all_options` was, with no benefits introduced at all.

oshtaier replied:
that still makes no sense to me
I have created new capsule using `vm.create()` command and now I want to run something on it. `vm.run` should take few seconds to execute, so not sure about logic mentioned

abalakh replied:
vm.run == ssh.command(hostname=vm.hostname), the difference is vm.run can _only_ be used in case the vm you're trying to point to was created by `vm.create` in the same test run previously; you can't pass any existing vm to save you hour of time.

oshtaier replied:
and why vm from setupclass is not in the same test run?

ldjebran commented on tests/foreman/api/test_contentmanagement.py:
```diff
@@ -148,6 +148,32 @@ def update_capsule_download_policy(self, capsule_id, download_policy):
+        )
+        self.assertEqual(result.return_code, 0)
+        # remove empty lines from output if any
+        result.stdout = [line for line in result.stdout if line]
```
why you are changing the result output you must create a new variable

ldjebran commented:
@abalakh think also that it's availability is enough here, as installing can lead to other problems for example dependency conflicts installation case must covered alone, but you can add an other test with install. 
so one test for availability and one test for install
@oshtaier what do you think?

abalakh commented:
@ldjebran other concerns are inability to re-execute such installation test for the same capsule and theoretically if that package breaks capsule in any way - it will affect all the rest tests from the module as the capsule is shared between them. That's why i tried to omit covering such case.

abalakh commented:
@ldjebran oh, and there's another issue with installing a package in a separate test - if such test will be executed _before_ package availability test, the last one will fail as the package will become not available to install. Imo even if we decide to test package installation, we'd better do it in the same test.

abalakh commented:
```python
pytest -v tests/foreman/api/test_contentmanagement.py -k test_positive_insights_puppet_package_availability
============================= test session starts ==============================
platform darwin -- Python 2.7.13, pytest-3.2.3, py-1.5.2, pluggy-0.4.0 -- /UsersAddressing comments
/andrii/workspace/env/bin/python
cachedir: .cache
rootdir: /Users/andrii/workspace/robottelo, inifile:
plugins: xdist-1.20.1, services-1.2.1, mock-1.6.3, forked-0.2, cov-2.5.1
collected 12 items
2017-12-13 10:39:53 - conftest - DEBUG - BZ deselect is disabled in settings


tests/foreman/api/test_contentmanagement.py::CapsuleContentManagementTestCase::test_positive_insights_puppet_package_availability PASSED

============================= 11 tests deselected ==============================
=================== 1 passed, 11 deselected in 6.09 seconds ====================
```
